### DISCUSSION TITLE
Troubleshooting srsRAN gNB-srsUE Project: Connection Issues and Validation Queries

### DISCUSSION THREAD

**Discussion by rajugbv** [srsran/srsRAN_Project#318] [General Help]
Hi everyone, I hope everyone is doing well.
 I am trying to build the 'srsRAN gNB with srsUE project.' For the gNB and UE, I am using USRP-2901. I was able to successfully establish a connection between the gNB and UE, but I am facing an error: RF status - O=0, U=1, L=0 after RRC NR reconfiguration successful. On the UE side, it has created the IP address 'tun_srsue' (10.45.0.15), but I cannot see the same IP address on the gNB side. Please check the screenshot below for details.
![image](https://github.com/srsran/srsRAN_Project/assets/41448193/d32b4f06-2b11-44c6-9cf3-aa4ca3a03336)
I can also ping and use iperf3 with the above IP address. I wanted to know how I can validate the results to ensure that I am getting correct results or not.

![image](https://github.com/srsran/srsRAN_Project/assets/41448193/2899691b-4ddf-4d79-81a5-c6983a41cc7e)
![image](https://github.com/srsran/srsRAN_Project/assets/41448193/1ed5c35d-2f8d-47b8-adaf-f90733daae0d)

below are the configuration please check and ask me if you need anything else

[ue_files.zip](https://github.com/srsran/srsRAN_Project/files/13380806/ue_files.zip)

[gnb-20231116T172644Z-001.zip](https://github.com/srsran/srsRAN_Project/files/13380739/gnb-20231116T172644Z-001.zip)

**Comment by brendan-mcauliffe** [comment]
Hi @rajugbv, your logs and pcaps look fine. I am not sure why you think there is an issue. 

Seeing a late now and then is not a major problem so long as the UE remains connected. I suggest you run the performance script, found [here](https://github.com/srsran/srsRAN_Project/tree/main/scripts), on both the UE and gNB machines. This may help reduce the number of lates. 

If you can successfully ping the Google DNS (8.8.8.8) from the UE, then you are connected to the gNB correctly and the traffic is being successfully routed through the core and out to the internet. The best way to ensure you are taking the necessary steps is to cross reference what you see on your own console output with that in the [tutorial](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html).

**Reply by rajugbv** [reply]
hey @brendan-mcauliffe brendan ,thank you replying i want to transmit a video using the about setup can you suggest a way?

**Reply by brendan-mcauliffe** [reply]
Yes, you can do this using VLC. Set up two instances, one for the UE and one for the gNB. Similarly to how iPerf works, one will act as a server, and one will act as a client. You can then choose a source media to stream, assign a destination to stream to and stream the video between the two nodes. Have a look at [this guide](https://www.androidauthority.com/vlc-media-player-stream-audio-video-3183564/) for a rough idea. I suggest you use UDP instead of HTTP for the transport protocol.
